src/wms/src/main/java/org/geoserver/wms/MapLayerInfo.java

```diff
@@ -257,12 +279,29 @@ public boolean isCachingEnabled() {
      *     0} if not set
      */
     public int getCacheMaxAge() {
-        if (layerInfo == null) {
-            return 0;
+        int maxAge = 0;
+
+        // maxAge may be deffined in override metadata, but caching
+        // may be disabled
+        if (this.isCachingEnabled(this.overrideMetadata)) {
+            maxAge = getCacheMaxAge(this.overrideMetadata);
         }
-        ResourceInfo resource = layerInfo.getResource();
-        Integer val = resource.getMetadata().get(ResourceInfo.CACHE_AGE_MAX, Integer.class);
-        return val == null ? 0 : val;
+
+        if (maxAge == 0) {
```

Comment:
So if caching is enabled in the group, but max age is set to zero, then we consult the single layer settings?
Then it's not an override, but something in between. 
I'm not 100% sure, but is this consistent with the behavior in the tile layer case? Does not seem like it is.

Reply:
Good point. Thanks. I changed it to return value from layer group metadata if enabled and do not continue to layer max age.